yt/data_objects/data_containers.py

```diff
@@ -808,11 +815,64 @@ def create_firefly_object(
             )
             datadir = JSONdir
 
+        if match_any_particle_types is None:
+            # not specified, switching to (temporary) default
+            issue_deprecation_warning(
+                "match_any_particle_types wasn't specified. Its current default "
+                "value (True) will be changed to False in the future. "
```

Comment:
In hindsight, I don't think we need to change the default in the future.
```suggestion
                "match_any_particle_types wasn't specified. "
```